src/map.cpp

```diff
@@ -8175,22 +8169,25 @@ template<typename Functor>
     }
 }
 
-void map::scent_blockers( bool (&blocks_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
-                          bool (&reduces_scent)[SEEX * MAPSIZE][SEEY * MAPSIZE],
+void map::scent_blockers( std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &blocks_scent,
+                          std::array<std::array<bool, SEEX * MAPSIZE>, SEEY * MAPSIZE> &reduces_scent,
```

Comment:
> I would move its content into the map cache system. 

This would make an additional coupling. You wouldn't be able to modify `scent_array` (which I consider an intimate implementation detail of `scent_map`) so easily in the future. So you can't just move the function there, you need a public interface to make this movement clean.